### PR TITLE
feat(ui): add organization server actions and scan launching

### PR DESCRIPTION
**Pull request by alejandrobailo**
### Context

PROWLER-1091 — PR 3/6 in the provider wizard chain. Base: PR 2 (#10154).

### Description

- **Organization server actions** (`actions/organizations/organizations.ts`): CRUD for organizations, organization secrets, discovery trigger/poll, apply discovery. All path parameters validated with `validatePathIdentifier` + `encodeURIComponent` (addresses SSRF scanner flags)
- **Tree adapter** (`organizations.adapter.ts`): Transforms flat API arrays (roots, OUs, accounts) into hierarchical `TreeDataItem[]`. Includes `getSelectableAccountIds`, `buildAccountLookup`, `getOuIdsForSelectedAccounts`
- **Scan launching** (`actions/scans/scans.ts`): `launchOrganizationScans` with `runWithConcurrencyLimit` (max 5 concurrent requests). `revalidatePath` called only on success
- **Tests**: Input validation rejection, conditional revalidation, concurrency ceiling verification, tree builder correctness

### Steps to review

1. **Security**: Check `validatePathIdentifier` in `organizations.ts` — rejects path traversal patterns
2. **Adapter**: Review `organizations.adapter.ts` — tree building, selectable account filtering
3. **Concurrency**: Review `launchOrganizationScans` — `runWithConcurrencyLimit` with worker pool pattern
4. **Tests**: Verify coverage of edge cases

#### UI
- [x] Path identifiers validated before URL construction
- [x] `revalidatePath` only on success
- [x] Concurrent scan launches capped at 5
- [x] Unit tests for all new functions

### Chain

| PR | Description | Status |
|----|-------------|--------|
| PR 1 | shadcn primitives + shared components | #10153 |
| PR 2 | Organization and wizard types + stores | #10154 |
| **→ PR 3** | Server actions and adapters | This PR |
| PR 4 | Provider wizard modal | Next |
| PR 5 | AWS Organizations flow | Pending |
| PR 6 | E2E test updates | Pending |

### License

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: ui/actions/scans/scans.ts [hunk 1]
```diff
@@ -13,6 +13,41 @@ import {
 } from "@/lib/provider-filters";
 import { addScanOperation } from "@/lib/sentry-breadcrumbs";
 import { handleApiError, handleApiResponse } from "@/lib/server-actions-helper";
+
+const ORGANIZATION_SCAN_CONCURRENCY_LIMIT = 5;
+
+export async function runWithConcurrencyLimit<T, R>(
```
Comment (Alan-TheGentleman) [inline ui/actions/scans/scans.ts]: two possible issues with runWithConcurrencyLimit: 

- Exported from a "use server" file, Next.js exposes it as a callable server action. It's a generic utility, I would move it to @/lib/concurrency.ts.
 
- At line 48, Promise.all(workers), if any worker throws uncaught, all partial results are lost. Either catch inside the utility or document that callers handle errors in their worker fn

File: ui/actions/organizations/organizations.ts [hunk 2]
```diff
@@ -0,0 +1,324 @@
...
+  }
+
+  if (!PATH_IDENTIFIER_PATTERN.test(normalizedValue)) {
+    return { error: invalidError };
+  }
+
+  return { value: normalizedValue };
+}
+
+function hasActionError(result: unknown): result is { error: unknown } {
```
Comment (Alan-TheGentleman) [inline ui/actions/organizations/organizations.ts:30]: hasActionError checks "error" in result and it would also match with { error: null }, just in case I would add a check for this situation

**Review by Alan-TheGentleman — DISMISSED** [review]
Only some comments

File: ui/actions/scans/scans.ts (hunk 1 above)
Comment (alejandrobailo) [reply ui/actions/scans/scans.ts]: Applied in commit 6f08d4d74. runWithConcurrencyLimit was moved out of the use-server file into ui/lib/concurrency.ts, and I added tests plus documentation clarifying uncaught worker-error behavior.

File: ui/actions/organizations/organizations.ts (hunk 2 above)
Comment (alejandrobailo) [reply ui/actions/organizations/organizations.ts:30]: Fixed in commit 6f08d4d74. hasActionError now ignores null/undefined error values so { error: null } is treated as non-error.
